processing/src/main/java/io/druid/query/filter/ValueGetter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.filter;
+
+/**
+ */
+public interface ValueGetter
+{
+  public String[] get();
```

Comment:
After this pull requests is merged it's a good idea to fix this. But right now it would result in a lot more code making the pull requests even harder to comprehend.

I think the only way to do this is in [`makeMatcher`](https://github.com/erikdubbelboer/druid/blob/3b59aa0d38a6a5958e1de0e00e53321539060abd/processing/src/main/java/io/druid/segment/filter/DimensionsFilter.java#L63) to see if all columns are integer or float. And then call a different [`makeValueMatcher`](https://github.com/erikdubbelboer/druid/blob/3b59aa0d38a6a5958e1de0e00e53321539060abd/processing/src/main/java/io/druid/segment/filter/DimensionsFilter.java#L81) that compares integers/floats instead of strings. This would mean the code of [`makeValueMatcher`](https://github.com/erikdubbelboer/druid/blob/3b59aa0d38a6a5958e1de0e00e53321539060abd/processing/src/main/java/io/druid/segment/filter/DimensionsFilter.java#L81) and [`overlap`](https://github.com/erikdubbelboer/druid/blob/3b59aa0d38a6a5958e1de0e00e53321539060abd/processing/src/main/java/io/druid/segment/filter/DimensionsFilter.java#L111) needs to be duplicated twice for integers and floats. That's another 106 lines of code in this file.